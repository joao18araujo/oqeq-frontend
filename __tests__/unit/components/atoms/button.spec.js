import React from 'react';
import { shallow } from 'enzyme';
import { TouchableHighlight, Text } from 'react-native';
import Button from '~components/atoms/button';

const initialProps = (props) => ({
  title: 'Test title',
  onPress: () => {},
  disabled: false,
  style: { color: '#0080ff' },
  textStyle: { backgroundColor: '#0080ff' },
  ...props,
});

describe('Button', () => {
  const onPressMock = jest.fn();
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallow(<Button {...initialProps({ onPress: onPressMock })} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should pass onPress, disabled and style to TouchableHighlight', () => {
    expect(wrapper.find(TouchableHighlight).props()).toEqual(
      expect.objectContaining({
        disabled: false,
        onPress: onPressMock,
        style: expect.arrayContaining([{ color: '#0080ff' }]),
      })
    );
  });

  it('should pass textStyle to Text', () => {
    expect(wrapper.find(Text).props()).toEqual(
      expect.objectContaining({
        style: expect.arrayContaining([{ backgroundColor: '#0080ff' }]),
      })
    );
  });

  it('should pass title to Text body', () => {
    expect(wrapper.find(Text).props().children).toEqual('Test title');
  });
});
