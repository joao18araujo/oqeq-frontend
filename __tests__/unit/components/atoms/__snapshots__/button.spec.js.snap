// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `"<View style=\\"0:[object Object];1:;2:[object Object]\\" focusable=\\"true\\"><Text style=\\"0:[object Object];1:[object Object]\\">Test title</Text></View>"`;
